AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -254,17 +254,81 @@ function Assert-IsNativeAD {
             -ErrorAction Stop
     }
 
+    $DirectoryServiceOptions = Get-DirectoryServiceOptions -StorageAccount $StorageAccount
+
+    if ("AD" -ne $DirectoryServiceOptions)
+    {
+        Write-Error -ErrorAction Stop -Message (
+            "The cmdlet is stopped due to the storage account '$StorageAccountName' having the DirectoryServiceOptions value: '$DirectoryServiceOptions'. " +
+            "The DirectoryServiceOptions for the account needs to be 'AD' in order to run the cmdlet."
+        )            
+    }

```

Comment:
Note that there is a change here in the behavior of `Assert-IsNativeAD`. It used to accept `null`, now it doesn't. We need to make sure that this is valid for all usages of `Assert-IsNativeAD`

Reply:
- [ ] `Update-AzStorageAccountADObjectPassword`
- [ ] `Invoke-AzStorageAccountADObjectPasswordRotation`
- [ ] `Update-AzStorageAccountAuthForAES256`

Reply:
Let me know if there's any feedback on the above

Reply:
I agree with this change.